pkg/kubelet/cm/container_manager_linux_test.go

```diff
@@ -343,3 +360,245 @@ func TestNewPodContainerManager(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerHasExclusiveCPUs(t *testing.T) {
+	guaranteedQOSResources := v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("1"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("1"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+	burstableQOSResources := v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("1"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+	guaranteedQOSResourcesWithNonIntegerCPU := v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("500m"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("500m"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+
+	testCases := []struct {
+		name                            string
+		pod                             *v1.Pod
+		containerName                   string
+		cpuSets                         map[string]cpuset.CPUSet
+		expectExclusiveCPUs             bool
+		podLevelResourceManagersEnabled bool
+	}{
+		{
+			name: "No pod-level resources, BestEffort container, has exclusive CPUs",
```

Comment:
can this case happens in practice? nothing warrants exclusive CPU allocation, so it seems a case created to exercise the existing code in an impossible way. We should never end up in this case in the first place, so there isn't really a behavior to expect here.

Reply:
Yes, this case is not right, changed it to be a Guaranteed container, which can actually happen.